File: core/weapons/gatlingGun.js
import { assets } from "../assets/main.js";
import { worldSize } from "../utils/constants.js";
import { randomNum } from "../utils/random.js";
import { wrap } from "../world/utils.js";
import Projectile from "./projectile.js";

export default class GatlingGun extends Projectile {
    constructor({ x, y, angle, ship, color, speed = 10 }) {
        super({
            name: "Gatling Gun",
            speed: speed,
            acceleration: 50000,
            x: x,
            y: y,
            width: 8,
            height: 15,
            angle: angle,
            damage: 20,
            range: 10000,
            fireRate: 0.5,
            energyCost: 50,
            ship,
            color,
            img: assets?.images?.projectile,
        });
    }

    update(t, dt) {
        this.angle += randomNum(-0.005, 0.005)
        super.update(t, dt);
    }
}

File: void_front/game/core/weapons/gatling_gun.py
from __future__ import annotations

import numpy as np
from game.core.weapons.projectile import Projectile


class GatlingGun(Projectile):
    def __init__(self, options, force=False):
        defaults = {
            "name": "Gatling Gun",
            "speed": options.get("speed", 10),
            "acceleration": 50000,
            "x": options.get("x"),
            "y": options.get("y"),
            "width": 8,
            "height": 15,
            "angle": options.get("angle"),
            "damage": 20,
            "range": 10000,
            "fireRate": 0.5,
            "energyCost": 50,
            "ship": options.get("ship"),
            "color": options.get("color"),
            "img": options.get("img"),
        }
        super().__init__(defaults, force)

    def update(self, t, dt):
        self.angle += np.random.uniform(-0.005, 0.005)
        super().update(t, dt)
